wire-ios-data-model/E2EIdentity/E2EIService.swift

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+public protocol E2EIServiceInterface {
+
+    func directoryResponse(directoryData: Data) throws -> WireCoreCrypto.AcmeDirectory
+    func getNewAccountRequest(previousNonce: String) async throws -> Data
+    func setAccountResponse(accountData: Data) async throws
+
+}
+
+public final class E2EIService: E2EIServiceInterface {
+
+    // MARK: - Properties
+
+    private weak var context: NSManagedObjectContext?
+    public var wireE2eIdentity: WireE2eIdentity?
+    private let coreCrypto: SafeCoreCryptoProtocol
+
+    // MARK: - Life cycle
+
+    public init(context: NSManagedObjectContext,
+                coreCrypto: SafeCoreCryptoProtocol) {
+        self.context = context
+        self.coreCrypto = coreCrypto
+    }
+
+    // MARK: - Setup enrollment
+
+    public func setupNewEnrollment() {
+        guard
+            let context = context,
+            let selfUserHandle = ZMUser.selfUser(in: context).handle,
+            let selfUserName = ZMUser.selfUser(in: context).name,
+            let selfClient = ZMUser.selfUser(in: context).selfClient(),
+            let clientId = MLSClientID(userClient: selfClient)
+        else {
+            return
+        }
+        // TODO: we should use the new CoreCrypto version: `e2eiNewRotateEnrollment` and `e2eiNewActivationEnrollment`
+        wireE2eIdentity = try? coreCrypto.perform { try $0.e2eiNewEnrollment(clientId: clientId.rawValue,
+                                                                             displayName: selfUserName,
+                                                                             handle: selfUserHandle,
+                                                                             expiryDays: UInt32(90),
+                                                                             ciphersuite: defaultCipherSuite.rawValue)
+
+        }
+
+    }
+
+    public func isE2EIEnabled() -> Bool {
+        // TODO: we should use the new CoreCrypto version
+        // return coreCrypto.e2eiIsEnabled(defaultCipherSuite.rawValue)
+        return true
+    }
+
+    // MARK: - WireE2EIdentity methods
+
+    public func directoryResponse(directoryData: Data) throws -> WireCoreCrypto.AcmeDirectory {
+        let buffer = [UInt8](directoryData)
```

Comment:
there's a `Data+Bytes` extension that does this, you can use it as below
```suggestion
        let buffer = directoryData.bytes
```

Reply:
@David-Henner I would actually question this. For me it hardly makes sense to define extensions/methods with a single line of code.

Reply:
there's a bit more to it than a simple line https://github.com/wireapp/wire-ios/blob/develop/wire-ios-utilities/Source/Public/Data%2BBytes.swift, but I agree

Reply:
I was referring to `Data.bytes`.
```
public extension Data {
    var bytes: [Byte] {
        return [Byte](self)
    }
```
Many projects have little helpers like this, but they don't add much value, in my opinion.

Reply:
I guess it's a personal thing, but I find reading the word "byte" more pleasant than `UInt8` and it slightly lessens the cognitive load.